packages/firestore/src/core/sync_engine.ts

```diff
@@ -227,7 +235,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
           const viewDocChanges = view.computeDocChanges(docs);
           const synthesizedTargetChange = TargetChange.createSynthesizedTargetChangeForCurrentChange(
             queryData.targetId,
-            current
+            current && this.onlineState !== OnlineState.Offline
```

Comment:
This is a improved version of the old code in `applyOnlineStateChange`. I am no longer marking a query non-current, but instead use the same logic to mark it non-current based on the online state.

I didn't like all the `this.isPrimary` check in `applyOnlineStateChange`.